RecoParticleFlow/PFClusterProducer/src/Basic2DGenericPFlowPositionCalc.cc

```diff
@@ -43,40 +53,37 @@ calculateAndSetPositionActual(reco::PFCluster& cluster) const {
 
   auto const recHitCollection = &(*cluster.recHitFractions()[0].recHitRef()) - cluster.recHitFractions()[0].recHitRef().key();
   auto nhits = cluster.recHitFractions().size();
-  struct LHit{ reco::PFRecHit const * hit; double energy; double fraction;};
+  struct LHit{ reco::PFRecHit const * hit; float energy; float fraction;};
   declareDynArray(LHit,nhits,hits);
   for(auto i=0U; i<nhits; ++i) {
     auto const & hf = cluster.recHitFractions()[i];
     auto k = hf.recHitRef().key();
     auto p = recHitCollection+k;
-    hits[i]= {p,(*p).energy(), hf.fraction()}; 
+    hits[i]= {p,(*p).energy(), float(hf.fraction())}; 
   }
 
-  if(_posCalcNCrystals != -1) // sorted to make neighbour search faster
-    std::sort(hits.begin(),hits.end(),[](LHit const& a, LHit const& b) { return a.hit<b.hit;});
 
-  LHit mySeed={nullptr}; 
+  bool resGiven = bool(_timeResolutionCalcBarrel) & bool(_timeResolutionCalcEndcap);
+  LHit mySeed={nullptr};
   for( auto const & rhf : hits ) {
     const reco::PFRecHit & refhit = *rhf.hit;
-    if( refhit.detId() == cluster.seed() ) mySeed = rhf;
-    const double rh_fraction = rhf.fraction;
-    const double rh_rawenergy = rhf.energy;
-    const double rh_energy = rh_rawenergy * rh_fraction;   
-    if( edm::isNotFinite(rh_energy) ) {
+    if( refhit.detId() == cluster.seed() )  mySeed = rhf;
+    const auto rh_fraction = rhf.fraction;
+    const auto rh_rawenergy = rhf.energy;
+    const auto rh_energy = rh_rawenergy * rh_fraction;   
+#ifdef PF_DEBUG
+    if unlikely( edm::isNotFinite(rh_energy) ) {
       throw cms::Exception("PFClusterAlgo")
 	<<"rechit " << refhit.detId() << " has a NaN energy... " 
 	<< "The input of the particle flow clustering seems to be corrupted.";
     }
+#endif
     cl_energy += rh_energy;
     // If time resolution is given, calculated weighted average
-    if ( bool(_timeResolutionCalcBarrel) & bool(_timeResolutionCalcEndcap) ) {
+    if ( resGiven ) {
       double res2 = 1.e-4;
       int cell_layer = (int)refhit.layer();
-      if (cell_layer == PFLayer::HCAL_BARREL1 ||
-          cell_layer == PFLayer::HCAL_BARREL2 ||
-          cell_layer == PFLayer::ECAL_BARREL)
-        res2 = 1./_timeResolutionCalcBarrel->timeResolution2(rh_rawenergy);
-      else
+      res2 =  isBarrel(cell_layer) ? 1./_timeResolutionCalcBarrel->timeResolution2(rh_rawenergy) :
         res2 = 1./_timeResolutionCalcEndcap->timeResolution2(rh_rawenergy);
```

Comment:
oops, clearly the second "res=" should not be there... ( I do not exclude that the result is correct though)
will fix